### PR TITLE
Add judgment before split the string

### PR DESCRIPTION
**Pull request by pengbinbin1**
Which issue(s) this PR fixes:
Without the judgment before split, the slices's length may less than 2, and the code _strings.Split(override, ":")[1]_ will panic

### CODE REVIEW

File: pkg/images/images.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func GetImagePath(repo, version, image, component string, imageOverrides map[str
 }
 
 func GetPullPolicy(version, override string) v1.PullPolicy {
-	if len(override) > 0 {
+	if len(override) > 0 && strings.Contains(override, ":") {
```
Comment (tpantelis) [inline pkg/images/images.go]: Just curious - did you observe an actual panic in a running system or did you just notice this on code inspection?

```suggestion
        tags := strings.Split(override, ":")
	if len(tags) > 1 {
	    return getPullPolicy(tags[1])
	}
	
	return getPullPolicy(version)
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/images/images.go (hunk 1 above)
Comment (pengbinbin1) [reply pkg/images/images.go]: Yes,I have observe an actual panic in my running system,like the below image
微信截图_20220908090401

File: pkg/images/images.go [hunk 2]
```diff
@@ -51,9 +51,11 @@ func GetImagePath(repo, version, image, component string, imageOverrides map[str
...
 
 func GetPullPolicy(version, override string) v1.PullPolicy {
-	if len(override) > 0 {
-		tag := strings.Split(override, ":")[1]
-		return getPullPolicy(tag)
+	if len(override) > 0
+		tags := strings.Split(override, ":")
+		if len(tags) > 1{
+			return getPullPolicy(tags[1])
+		}
```
Comment (pengbinbin1) [inline pkg/images/images.go]: fixed

File: pkg/images/images.go [hunk 3]
```diff
@@ -51,9 +51,11 @@ func GetImagePath(repo, version, image, component string, imageOverrides map[str
 }
 
 func GetPullPolicy(version, override string) v1.PullPolicy {
-	if len(override) > 0 {
-		tag := strings.Split(override, ":")[1]
-		return getPullPolicy(tag)
+	if len(override) > 0
```
Comment (tpantelis) [inline pkg/images/images.go]: You don't need this check - `Split` handles empty string.

```suggestion
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/images/images.go (hunk 3 above)
Comment (pengbinbin1) [reply pkg/images/images.go]: I think we need this check,if the string is empty then we will unnecessary to split it even though empty string can also be splited

Comment (tpantelis) [reply pkg/images/images.go]: As I mentioned, `Split` handles an empty string and returns an empty array, ie there's nothing to split. This is a bit cleaner than having a redundant check.

Comment (pengbinbin1) [reply pkg/images/images.go]: Done

**Review by tpantelis — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**